fe/src/main/coffee/channels.coffee

```diff
@@ -0,0 +1,14 @@
+
+validChannelHandler = (channels, pathname, cb) ->
+  [_..., modeHint] = pathname.split '/'
+  @mode = if modeHint in channels
+    cb modeHint
+  else
+    $ -> document.write "You are not supposed to be on this page."
```

Comment:
This is more intended as a precaution against mistyping the address than anything else